datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1308,16 +1310,16 @@ fn ensure_distribution(
                 // When `repartition_file_scans` is set, leverage source operators
                 // (`ParquetExec`, `CsvExec` etc.) to increase parallelism at the source.
                 if repartition_file_scans {
+                    #[cfg(feature = "parquet")]
```

Comment:
It would be nice to remove this `#[cfg` as well as the special case handling of ParquetExec in the enforce distribution rule. Not in this PR, but as a follow on. I will file a ticket